streaming/src/main/scala/org/apache/spark/streaming/scheduler/ReceiverTracker.scala

```diff
@@ -477,7 +493,12 @@ class ReceiverTracker(ssc: StreamingContext, skipReceiverLaunch: Boolean = false
           registerReceiver(streamId, typ, hostPort, receiverEndpoint, context.senderAddress)
         context.reply(successful)
       case AddBlock(receivedBlockInfo) =>
-        context.reply(addBlock(receivedBlockInfo))
+        if (WriteAheadLogUtils.isBatchingEnabled(ssc.conf)) {
+          val f = Future(addBlockAsync(receivedBlockInfo))(walBatchingThreadPool)
+          f.onComplete(result => context.reply(result.get))(walBatchingThreadPool)
+        } else {
+          context.reply(addBlock(receivedBlockInfo))
```

Comment:
We didn't want to change the existing codepath. That is practically the bottleneck